paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSource.java

```diff
@@ -53,11 +68,43 @@ public SplitEnumerator<FileStoreSourceSplit, PendingSplitsCheckpoint> restoreEnu
             PendingSplitsCheckpoint checkpoint) {
         Collection<FileStoreSourceSplit> splits =
                 checkpoint == null ? getSplits() : checkpoint.splits();
-        return new StaticFileStoreSplitEnumerator(context, null, splits, splitBatchSize);
+        SplitAssigner splitAssigner =
+                createSplitAssigner(context, splitBatchSize, splitAssignMode, splits);
+        return new StaticFileStoreSplitEnumerator(context, null, splitAssigner);
     }
 
     private List<FileStoreSourceSplit> getSplits() {
         FileStoreSourceSplitGenerator splitGenerator = new FileStoreSourceSplitGenerator();
         return splitGenerator.createSplits(readBuilder.newScan().plan());
     }
+
+    public static SplitAssigner createSplitAssigner(
+            SplitEnumeratorContext<FileStoreSourceSplit> context,
+            int splitBatchSize,
+            SplitAssignMode splitAssignMode,
+            Collection<FileStoreSourceSplit> splits) {
+        switch (splitAssignMode) {
+            case FAIR:
+                return new FairSplitAssigner(
+                        splitBatchSize,
+                        createSplitAssignment(splits, context.currentParallelism()));
+            case PREEMPTIVE:
+                return new PreemptiveSplitAssigner(new LinkedList<>(splits));
+            default:
+                throw new UnsupportedOperationException(
+                        "Unsupported assign mode " + splitAssignMode.toString());
+        }
+    }
+
+    private static Map<Integer, LinkedList<FileStoreSourceSplit>> createSplitAssignment(
```

Comment:
move this method into FairSplitAssigner. It is a part of `FAIR`.

Reply:
resolved.